File: app/src/main/java/com/cyberIyke/allvideodowloader/adapters/SuggestionAdapter.java
package com.cyberIyke.allvideodowloader.adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.cyberIyke.allvideodowloader.R;
import com.cyberIyke.allvideodowloader.webservice.Result;

import java.util.ArrayList;
import java.util.List;

public class SuggestionAdapter extends RecyclerView.Adapter<SuggestionAdapter.ViewHolder> {
    List<Result> resultList = new ArrayList<>();
    SuggetionListner suggetionListner;

    public SuggestionAdapter(SuggetionListner suggetionListner) {
        this.suggetionListner = suggetionListner;
    }

    public void setResultList(List<Result> resultList) {
        this.resultList = resultList;
        if (this.resultList == null){
            this.resultList = new ArrayList<>();
        }
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_search_suggestion, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Result result = resultList.get(position);

        if (result != null && result.getKey() != null) {
            holder.txtTitle.setText(result.getKey());
            String strSubTitle = "Search for \"" + result.getKey() + "\"";
            holder.txtSubTitle.setText(strSubTitle);

            holder.itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    if (suggetionListner != null) {
                        suggetionListner.onSuggetion(result.getKey());
                    }
                }
            });
        }
    }

    @Override
    public int getItemCount() {
        return resultList.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        TextView txtTitle, txtSubTitle;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            txtTitle = itemView.findViewById(R.id.txtTitle);
            txtSubTitle = itemView.findViewById(R.id.txtSubTitle);
        }
    }

    public interface SuggetionListner {
        void onSuggetion(String str);
    }
}

File: app/src/main/java/com/cyberIyke/allvideodowloader/adapters/SuggestionAdapter.kt
package com.cyberIyke.allvideodowloader.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.cyberIyke.allvideodowloader.R
import com.cyberIyke.allvideodowloader.webservice.Result

class SuggestionAdapter constructor(var suggetionListner: SuggetionListner?) :
    RecyclerView.Adapter<SuggestionAdapter.ViewHolder>() {

    private var resultList: List<Result>? = ArrayList()

    fun resultList(resultList: List<Result>?) {
        this.resultList = resultList
        if (this.resultList == null) {
            this.resultList = ArrayList()
        }
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): SuggestionAdapter.ViewHolder {
        return ViewHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.item_search_suggestion, parent, false)
        )
    }

    override fun onBindViewHolder(holder: SuggestionAdapter.ViewHolder, position: Int) {
        val result: Result = resultList!!.get(position)
        if (result.key != null) {
            holder.txtTitle.text = result.key
            val strSubTitle: String = "Search for \"" + result.key + "\""
            holder.txtSubTitle.text = strSubTitle
            holder.itemView.setOnClickListener {
                if (suggetionListner != null) {
                    suggetionListner!!.onSuggetion(result.key)
                }
            }
        }
    }

    override fun getItemCount(): Int {
        return resultList!!.size
    }

    inner class ViewHolder constructor(itemView: View) :
        RecyclerView.ViewHolder(itemView) {
        var txtTitle: TextView
        var txtSubTitle: TextView

        init {
            txtTitle = itemView.findViewById(R.id.txtTitle)
            txtSubTitle = itemView.findViewById(R.id.txtSubTitle)
        }
    }

    open interface SuggetionListner {
        fun onSuggetion(str: String?)
    }
}